### PR TITLE
:tada: Source Hubspot: Add private app authentication

### PR DESCRIPTION
**Pull request by vladimir-remar**
## What
Related to this [upcoming-api-key-sunset](https://developers.hubspot.com/changelog/upcoming-api-key-sunset)

## How
Add a ```TokenAuthenticator```

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [X] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [X] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by harshithmullapudi** [comment]
Hey @vladimir-remar thanks for the contribution. Couple of more steps and we are there. Here is documentation for updating an existing connector https://docs.airbyte.io/connector-development#updating-an-existing-connector

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py [hunk 1]
```diff
@@ -87,7 +87,8 @@ def get_common_params(self, config) -> Mapping[str, Any]:
         api = self.get_api(config=config)
         common_params = dict(api=api, start_date=start_date, credentials=credentials)
 
-        if credentials.get("credentials_title") == "OAuth Credentials":
```
Comment (harshithmullapudi) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/source.py:90]: can we create constants for 2 strings and use them instead?

**Comment by harshithmullapudi** [comment]
Hey @vladimir-remar could you look at that one small change?

**Comment by vladimir-remar** [comment]
> Hey @vladimir-remar could you look at that one small change?
Thanks for your patience, I will check it ASAP.

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 2]
```diff
@@ -113,6 +113,11 @@ def is_oauth2(self) -> bool:
 
         return credentials_title == "OAuth Credentials"
 
+    def is_private_app(self) -> bool:
+        credentials_title = self.credentials.get("credentials_title")
+
+        return credentials_title == "Private App Credentials"
```
Comment (bjgbeelen) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:119]: I was looking for this feature and glad that I found a PR _hot of the press_

I was interested about the code change required for this feature and while looking at the code I thought I could do this very small suggestion, i hope you don't mind. There already was a comment about using a constant for the hardcoded strings. It's been replaced at one location, but here we find a similar situation.

Alternatively we could move the constants to a `constants.py` file and import the constants at both locations where they are needed. Also indicates that there is a relation between the two files.

Thanks for all effort so far, looking forward to getting this merged!

**Comment by vladimir-remar** [comment]
@harshithmullapudi could you take a look on this and let me know if there is something new to update?

**Comment by bjgbeelen** [comment]
@vladimir-remar I see the branch has conflicts again, that's something to already resolve perhaps for smooth merging soon

**Comment by vladimir-remar** [comment]
> @vladimir-remar I see the branch has conflicts again, that's something to already resolve perhaps for smooth merging soon

@bjgbeelen @harshithmullapudi conflicts already fixed

File: airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py [hunk 3]
```diff
@@ -129,7 +137,7 @@ def __init__(self, credentials: Mapping[str, Any]):
         self.credentials = credentials
         credentials_title = credentials.get("credentials_title")
 
-        if self.is_oauth2():
+        if self.is_oauth2() or self.is_private_app():
             self._session.auth = self.get_authenticator()
         elif credentials_title == "API Key Credentials":
```
Comment (faacsousa) [inline airbyte-integrations/connectors/source-hubspot/source_hubspot/streams.py:142]: Can we add "API Key Credentials" to constants.py? For consistency

**Comment by optical** [comment]
Hi folks, apologies if this isn't the correct place for this question. Looking at using the Private App authentication here, but I'm unsure of exactly which Private App scopes are required by the integration? It looks like they're different to the oauth scopes that are well documented in the Airbyte Hubspot connector docs

**Pull request merged**